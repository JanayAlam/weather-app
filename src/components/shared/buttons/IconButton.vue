<template>
  <v-btn
    @click="onClickHandler"
    :class="btnClasses"
    density="comfortable"
    icon=""
    :variant="variant"
  >
    <v-icon :icon="icons || ''" :class="classes + ' icon'" />
    <v-tooltip
      v-if="title"
      :text="title || ''"
      activator="parent"
      :location="tooltipLocation"
    />
  </v-btn>
</template>

<script setup>
const props = defineProps({
  onClickHandler: {
    type: Function,
    required: true,
  },
  title: {
    type: String,
    required: false,
  },
  classes: {
    type: String,
    default: '',
  },
  btnClasses: {
    type: String,
    default: '',
  },
  icons: {
    type: String,
    required: true,
  },
  tooltipLocation: {
    type: String,
    default: 'bottom',
  },
  variant: {
    type: String,
    default: 'tonal',
  },
});
</script>

<style scoped>
.icon {
  font-size: 1.3rem;
}
</style>
