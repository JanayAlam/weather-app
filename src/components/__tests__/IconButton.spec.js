import { mount } from '@vue/test-utils';
import { describe, expect, it } from 'vitest';
import IconButton from '../shared/buttons/IconButton.vue';

describe('IconButton', () => {
  const wrapper = mount(IconButton, {
    propsData: {
      onClickHandler: () => {},
      title: 'This is title',
      icons: 'fas fa-trash',
    },
  });

  it('should receive right props', async () => {
    expect(IconButton).toBeTruthy();
    expect(wrapper.html()).toMatchSnapshot();

    const props = wrapper.props();

    expect(props.title).toBe('This is title');
    expect(props.icons).toBe('fas fa-trash');
    expect(props.onClickHandler).toBeTruthy();
  });

  it('should mount all elements', () => {
    expect(wrapper.find('#icon-button-btn')).toBeTruthy();
    expect(wrapper.find('#icon-button-icon')).toBeTruthy();
    expect(wrapper.find('#icon-button-tooltip')).toBeTruthy();
  });
});
