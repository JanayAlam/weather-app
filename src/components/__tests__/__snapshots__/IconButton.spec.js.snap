// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`IconButton > should receive right props 1`] = `
"<v-btn data-v-c0cfd068="" class="" density="comfortable" icon="" variant="tonal" id="icon-button-btn">
  <v-icon data-v-c0cfd068="" icon="fas fa-trash" class="icon" id="icon-button-icon"></v-icon>
  <v-tooltip data-v-c0cfd068="" id="icon-button-tooltip" text="This is title" activator="parent" location="bottom"></v-tooltip>
</v-btn>"
`;
